http/typescript/fetch/src/httpCore.ts

```diff
@@ -3,98 +3,118 @@ import { Headers as FetchHeadersCtor } from 'cross-fetch';
 import { ReadableStream } from 'web-streams-polyfill';
 import { URLSearchParams } from 'url';
 import { HttpClient } from './httpClient';
+import { Middleware } from "./middleware";
+import { FetchOptions } from './fetchOptions';
+
+export interface RequestBuilderConfig {
+    parseNodeFactory: ParseNodeFactory,
+    serializationWriterFactory: SerializationWriterFactory
+}
+
+export interface CoreConfig {
```

Comment:
this interface and CoreOptions overlap a lot, is there a way we could merge them?

Reply:
The CoreOptions is internal will contain the non optional parameters such as HttpClient after we parse and set the properties.

Reply:
I get the subtilty now, thanks for explaining that. Can I suggest using doc comments on those two interfaces to specific such a difference? that one is used to init the service, the other one is a configuration bearer?